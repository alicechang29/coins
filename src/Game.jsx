import React, { useState } from "react";
//import "./Game.css";
import Coin from "./Coin.jsx";
import { getCoin } from "./coin.js";

/** Play coin flipping game
 *
 * State:
 * - coin: {face}
 * - counter: {heads, tails}
 */

function Game() {
  console.log("Game");

  const initialCoin = {
    caption: "",
    src: ""
  };


  const [coin, setCoin] = useState(initialCoin);
  const [headCount, setHeadCount] = useState(0);
  const [tailCount, setTailCount] = useState(0);

  // const [count, setCount] = useState(0);

  // function incrCount() {
  //   setCount(count => count + 1);
  // }


  function flipCoin() {
    console.log("flipCoin");
    const coin = getCoin();
    console.log("COIN", coin);
    setCoin(coin);

    if (coin.caption === "Heads") {
      setHeadCount(oldCount => oldCount + 1);
    } else if (coin.caption === "Tails") {
      setTailCount(oldCount => oldCount + 1);
    }
  }

  return (
    <div className="Game">
      <h1>Let's flip a coin</h1>
      <Coin
        caption={coin.caption}
        src={coin.src}
      />
      <button onClick={flipCoin}>Flip Coin</button>
      <p>Out of {tailCount + headCount} flips, there have
        been {headCount} heads and {tailCount} tails. </p>
    </div>
  );
}

export default Game;