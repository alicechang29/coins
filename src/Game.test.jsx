import { it, expect, vi, beforeEach, afterEach } from "vitest";

import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Game from "./Game";
// import TEST_IMAGES from "./_testCommon.js";

Math.random = vi.fn();

beforeEach(function () {
  Math.random
    .mockReturnValueOnce(0.25)
    .mockReturnValueOnce(0.75);
});

afterEach(function () {
  Math.random.mockReset();
});

it("renders without crashing", function () {
  render(<Game />);
});

it("matches snapshot", function () {
  const { container } = render(<Game />);
  expect(container).toMatchSnapshot();
});
