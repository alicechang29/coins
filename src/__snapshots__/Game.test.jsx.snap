// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Game"
  >
    <h1>
      Let's flip a coin
    </h1>
    <div
      class="Coin"
    >
      <h4
        class="Coin-title"
      />
      <img
        alt=""
        class="Coin-image"
        src=""
      />
    </div>
    <button>
      Flip Coin
    </button>
    <p>
      Out of 
      0
       flips, there have been 
      0
       heads and 
      0
       tails. 
    </p>
  </div>
</div>
`;
